### PR TITLE
[5.x] Add support for new google authentication library

### PR DESCRIPTION
**Pull request by CosminBd**
Created a new provider which uses the new Google Authentication Library (https://developers.google.com/identity/gsi/web/guides/overview?hl=en) 

This is related to #634.

To use the new library you would have to use `Socialite::driver('google-jwt')->user()`

the redirect method is deprecated, as google allows only for HTML and JS integrations, and now throws a message. 

For client side one could use:

 ```
<script src="https://accounts.google.com/gsi/client" async defer></script>

                    <div id="g_id_onload"
                         data-client_id="{{config('services.google.client_id')}}"
                         data-login_uri="{{config('services.google.redirect')}}"
                         data-auto_prompt="false">
                    </div>
                    <div class="g_id_signin"
                         data-type="standard"
                         data-size="large"
                         data-theme="outline"
                         data-text="sign_in_with"
                         data-shape="rectangular"
                         data-logo_alignment="left">
                    </div>
```

Basically, Google moved away from OAuth and now issue JWT's in order to authenticate the user.

My PR checks for the JWT, validates it and returns an User instance based on data provided.
I've also added a dependency for the openssl PHP extension, which is used to verify the signature of the JWT's with Google certs.

Both authentication methods are available via: 'google' and 'google-jwt' drivers.

I didn't add testing yet because I didn't know which package to use for mocking the JWT payloads.

### CODE REVIEW

**Comment by nunomaduro** [comment]
@CosminBd Regarding your question on Twitter: By "Laravel coding standards", Dries only means that you should format the code exactly as typically is formatted on other places on the framework regarding PHP annotations, etc.  Here is an example:

```diff
class GoogleProvider implements Provider
{
+   /**
+    * Create a new google provider instance.
+    *
+    * @param  \Illuminate\Http\Request  $request
+    * @param  string  $clientId
+    */
    public function __construct(Request $request, $clientId)
    {
        $this->request = $request;
        $this->clientId = $clientId;
    }
```

Another example:

```diff
    /**
     * Verifies the signature of the JWT issued, via Google Certs
     *
     * @param  string  $jwt
+    * @return bool
-    * @return boolean
     */
-   protected function verifyJwtSignature($jwt){
+   protected function verifyJwtSignature($jwt)
+   {
-       $jwt_parts = explode('.', $jwt);
+       $jwtParts = explode('.', $jwt);
          // ...
```

**Comment by nunomaduro** [comment]
@CosminBd Regarding testing, I would probably take a look about how currently things are being tested on other/existing providers, and try to follow the existing examples regarding testing.

**Comment by taylorotwell** [comment]
Can be a community maintained driver.

**Comment by codespearhead** [comment]
@CosminBd It seems that [SocialiteProviders/Providers](https://github.com/SocialiteProviders/Providers/issues/940#issuecomment-1858690586) is willing to accept it there.
